textproc/par/Portfile

```diff
@@ -18,26 +19,31 @@ master_sites        ${homepage}
 distname            [string totitle $name][string map {. {}} $version]
 
 checksums           rmd160  253070f319a7f3b3a6f0dad5797882efdbf597bc \
-                    sha256  33dcdae905f4b4267b4dc1f3efb032d79705ca8d2122e17efdecfd8162067082
+                    sha256  33dcdae905f4b4267b4dc1f3efb032d79705ca8d2122e17efdecfd8162067082 \
+                    size    47999
 
 patchfiles          patch-nbspace.diff \
                     patch-ptrdiffwarn.diff \
-                    patch-spelling.diff
-
-use_configure       no
+                    patch-spelling.diff \
+                    patch-protoMakefile.diff
 
 default_variants    +i18n
 variant i18n description adds support for multibyte characters {
     patchfiles-append   patch-i18n.diff
 }
 
-build.args          -f protoMakefile \
-                    CC="${configure.cc} \
-                        -c ${configure.optflags} [get_canonical_archflags cc]" \
-                    LINK1="${configure.cc} [get_canonical_archflags ld]"
+makefile.override   CC CFLAGS LDFLAGS PREFIX MANPREFIX
+
 build.target
 
+post-patch {
+    file copy "${worksrcpath}/protoMakefile" "${worksrcpath}/Makefile"
+}
+
```

Comment:
I realise this was previously covered by `build.args   -f protoMakefile`, however, when I was doing the upgrade to 1.53.0 it had a test phase that would also need this. Therefore it seemed less awkward to rename than to reference it.

Happy to do it more like we have here too though: https://github.com/macports/macports-ports/blob/28a01613446ef5e25dbdf51b5fd65084e2dfbb65/sysutils/entr/Portfile#L32-L34

Reply:
Since the port doesn't enable the test phase (i.e. doesn't say `test.run yes`) and overrides the destroot phase, it only needs the `-f` flag in `build.args` so that would be preferable to copying the Makefile needlessly.

Even if the port did need the flag in several phases, it takes but a moment to write the code that copies e.g. `build.args` to `destroot.args` (which would be `destroot.args {*}${build.args}`).

Reply:
Added in 61c6e2f.

For future reference, what does the the `{*}` do at the beginning of `{*}${build.args}`?

Reply:
It is the [expand operator](https://wiki.tcl-lang.org/page/%7B%2A%7D).

`destroot.args`, like all MacPorts options, is a list.

1. `destroot.args a b` sets it to a two-item list: a b.
2. `set mylist [list a b]` `destroot.args $mylist` would not be the same as (1): it would set `destroot.args` to a one-item list, and that one item is a two-item list a b.
3. To accomplish the same thing as (1) with a variable you would need to expand the list `mylist` into each of its elements using the expand operator: `destroot.args {*}$mylist`. Now `destroot.args` is a two-item list a b.